src/signMessage.c

```diff
@@ -64,8 +64,8 @@ void handleSignMessage(uint8_t p1, uint8_t p2, uint8_t *dataBuffer, uint16_t dat
     cx_ecfp_private_key_t privateKey;
     derive_private_key(&privateKey, derivationPath, derivationPathLength);
 
-    int messageLength = dataBuffer[0];
-    dataBuffer += 1;
+    int messageLength = (dataBuffer[0] << 8) + dataBuffer[1];
```

Comment:
Big Endian, eh? Is that what others are doing for ADPU messages?

Reply:
Just realized we are also size limited here: https://github.com/solana-labs/ledger-app-solana/blob/afb7eb2201161459dfa70c8e8777130002f354fc/src/main.c#L63 (`G_io_apdu_buffer[OFFSET_LC]`), so more work needed.

I haven't done a real exhaustive search of other ledger apps on the BE vs LE point (most are accepting data lengths <256), but we are using BE for the key derivation path already, a la Stellar: https://github.com/solana-labs/ledger-app-solana/blob/afb7eb2201161459dfa70c8e8777130002f354fc/src/getPubkey.c#L20

Reply:
Okay, updated, and added long-message signing test.